src/item_group.cpp

```diff
@@ -575,7 +575,7 @@ void Item_modifier::modify( item &new_item, const std::string &context ) const
                     mag.ammo_set( mag.ammo_default(), ch );
                 }
                 new_item.put_in( mag, item_pocket::pocket_type::MAGAZINE_WELL );
-            } else if( new_item.is_magazine() ) {
+            } else if( new_item.is_magazine() || new_item.magazine_current() ) {
```

Comment:
```suggestion
            } else if( new_item.is_magazine() ) {
```
This code doesn't belong here